src/workerd/io/worker-interface.capnp

```diff
@@ -147,6 +186,277 @@ struct Trace @0x8e8d911203762d34 {
+    outcome @0 :EventOutcome;
+
+    # Any additional arbitrary metadata that should be associated with the outcome.
+    tags @1 :List(Tag);
+  }
+
+  # An outcome information object that describes additional detail about the outcome
+  # of an Actor/Durable object. Used primarily to identify the ActorFlushReason.
+  struct ActorFlushInfo {
+    enum Common {
+      reason @0;
+      broken @1;
+    }
+    tags @0 :List(Tag);
+  }
+
+  # A detail event indicating a subrequest that was made during a request. This
+  # can be a fetch subrequest, an RPC subrequest, a call out to a KV namespace,
+  # etc.
+  # TODO(streaming-trace): This needs to be flushed out more.
+  struct Subrequest {
+    # A monotonic sequence number that is unique within the tail. The id is
+    # used primarily to correlate with the SubrequestOutcome.
+    id @0 :UInt32;
+    info :union {
+      none @1 :Void;
+      fetch @2 :FetchEventInfo;
+      jsRpc @3 :JsRpcEventInfo;
+    }
+  }
+
+  struct SubrequestOutcome {
+    id @0 :UInt32;
+    info :union {
+      none @1 :Void;
+      fetch @2 :FetchResponseInfo;
+      custom @3 :List(Tag);
+    }
+    # A Subrequest is really a specialist kind of span, so it can have an outcome.
+    # just like a span. Unlike regular spans tho, they cannot be transactional,
+    # and are not part of the normal span stack.
+    outcome @4 :Span.SpanOutcome;
+  }
+
+  # A span event is sent only at the completion of a span, and includes markers
+  # for the start and end times, as well as the start and end sequence numbers.
+  # A span event always occurs in the scope of the parent span (or the null span
+  # if there is no current). So, for example:
+  #
+  #  Event 0 - Onset (null span)
+  #  Event 1 - Info (null span)
+  #  Event 2 - Log (Span 1)  (new span implicitly started)
+  #  Event 3 - Log (Span 2)  (new span implicitly started)
+  #  Event 4 - Span 2 (Span 1)  (span 2 is complete, span 1 is still open)
+  #  Event 5 - Span 1 (null span)  (span 1 is complete)
+  #  Event 6 - Outcome (null span)
+  #
+  # Note that the Span 2 event occurs within Span 1
+  struct Span {
+    id @0 :UInt32;
+    parent @1 :UInt32;
+    transactional @2 :Bool;
+
+    # A span event may have an outcome field. If, for instance, the span represents
+    # events occuring while an output gate is open, and the output gate fails indicating
+    # that the events are not longer valid, the outcome field will be used to signal.
+    enum SpanOutcome {
+      unknown @0;
+      ok @1;
+      exception @2;
+      canceled @3;
+    }
+
+    outcome @3 :SpanOutcome;
+
+    # The info field is used to provide additional information about the outcome.
+    # These are often dependent on the type of info event that started the span.
+    # This is most typically only used for stage spans.
+    # For instance, a fetch event will have a fetchResponse info field in the outcome.
+    info :union {
+      none @4 :Void;
+      fetch @5 :FetchResponseInfo;
```

Comment:
This seems to be present in both Span and SubrequestOutcome. Which one is actually used to signal a fetch outcome?

Reply:
`Span` is used, in part, to indicate the close of an incoming request. `SubrequestOutcome` is used only to indicate the result of an outgoing subrequest. Consider `Subrequest` and `SubrequestOutcome` to be a simplified form of span representing only outbound subrequests.